Userland/Applications/HexEditor/HexEditorWidget.cpp

```diff
@@ -402,6 +403,9 @@ void HexEditorWidget::update_inspector_values(size_t position)
 
     // FIXME: Parse as other values like Timestamp etc
 
+    DeprecatedString utf16_string = TextCodec::decoder_for("utf-16le"sv)->to_utf8(StringView(selected_bytes.span()));
+    value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UTF16String, utf16_string);
```

Comment:
What happens here if the UTF-16 is not valid?

Reply:
Hmm it seems like it gives me garbled/garbage characters. But I haven't studied UTF-16 well enough to verify that these characters are indeed garbage and not valid

This seems like another issue. What happens if they are invalid? Seems like we should show `<?>` glyphs for each invalid character. There are weird edge cases when we select, say, a single byte. Should we show a `<?>` character, or nothing at all...? 🤔

Reply:
> Hmm it seems like it gives me garbled/garbage characters. Though I haven't studied UTF-16 well enough to verify that these characters are indeed garbage or whether they're actually valid

Right, it is much safer to show an "invalid" character or some sort of meta "invalid" message instead of garbage data.

> Another issue is what happens if they are invalid? Seems like we should show `<?>` glyphs for each invalid character, not a garbled character

If a single UTF-16 two-byte sequence is invalid, we should ideally show some replacement character which signals "invalid encoding" to the user instead of just "valid encoding for absent code point". However we have no useful way of doing that right now, so I guess � is fine for now. We should definitely not invalidate the entire string if possible, as that is not useful information; imagine the user trying to figure out which part of a long string in a binary file is invalid.

> There are weird edge cases though, when we select, say, a single byte. Should we show a `<?>` character, or no character at all...? 🤔

A single byte is never valid UTF-16, so we can show nothing at all. It's the same as with too few bytes for the multi-byte number types.

Reply:
> Right, it is much safer to show an "invalid" character or some sort of meta "invalid" message instead of garbage data.

👍

Going back to above, I take back what I said about garbled characters. I cannot (in fact) verify that garbled data is produced. That is because the only code points in the UTF-16 2 byte range are `U+D800` to `U+DFFF` , and these decode to `<?>` (expected behavior). Any other "weird" characters I get by randomly typing in 2 bytes pairs I assume are, in fact, valid UTF-16 decoded characters (until I see otherwise)

> absent code point

By “absent code point”, do you mean “code point corresponding to a character that serenity doesn’t know how to display”?

> imagine the user trying to figure out which part of a long string in a binary file is invalid

So it’s kinda difficult… the UTF-16 decoder is resilient when I insert an invalid (2 byte) character in the middle of a string of (2 byte) characters (e.g `$$<?>$`). However, if I append a null byte in front of a sequence of valid UTF-16 characters, the whole thing is decoded incorrectly

Seems like this is getting into the general problem of decoding UTF-16, what’s valid, what’s not, which seems really hard. I’d prefer to leave that logic in the UTF-16 decoder for now, and wait until we have some issues and take those case by case to see if we should deviate from the UTF-16 decoder and something custom in HexEditor

Reply:
> A single byte is never valid UTF-16, so we can show nothing at all. It's the same as with too few bytes for the multi-byte number types.

Following up, I confirmed that nothing is diplayed in the UTF-16 String view when a single byte is highlighted